dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java

```diff
@@ -1326,7 +1327,7 @@ private void setConnectionSpecificInfo( DatabaseMeta meta ) {
     getControls();
 
     if ( databaseDialectList != null ) {
-      databaseDialectList.setElements( databaseDialects  );
+      databaseDialectList.setElements( Arrays.asList( databaseDialects.toArray() )  );
```

Comment:
Would it work to use Collections.sort on the ArrayList instead of using a TreeSet?

Reply:
Let my try that